_posts/2019-11-27-gettingstarted-2020/getting-started-2020.Rmd

```diff
@@ -0,0 +1,144 @@
+
+In current Keras, two mechanisms are central to data preparation. One is the use of [tfdatasets pipelines](https://tensorflow.rstudio.com/guide/tfdatasets/introduction/). `tfdatasets` lets you load data in a streaming fashion (batch-by-batch), optionally applying transformations as you go. The other handy device here is [feature specs](https://tensorflow.rstudio.com/guide/tfdatasets/feature_spec/) resp. [feature columns](https://tensorflow.rstudio.com/guide/tfdatasets/feature_columns/). Together with a matching Keras layer, these allow for transforming the input data without having to think about what the new format will mean to Keras.
+
+While there are other types of data not discussed in the docs, the principles -- pre-processing pipelines and feature extraction -- generalize.
+
+### Model saving
+
+The best-performing model is of little use if ephemeral. Straightforward ways of saving Keras models are explained in a dedicated [tutorial](https://tensorflow.rstudio.com/tutorials/beginners/basic-ml/tutorial_save_and_restore/). 
+
+<aside>
+Advanced users: Additional options exist; see the [tutorial on checkpoints](https://tensorflow.rstudio.com/guide/saving/checkpoints/).
+</aside>
+
+And unless one's just tinkering around, the question will often be: How can I deploy my model?
+There is a complete new section on [deployment](https://tensorflow.rstudio.com/deploy/), featuring options like `plumber`, Shiny, TensorFlow Serving and RStudio Connect.
+
+<aside>
+Check out the new section on [deployment options](https://tensorflow.rstudio.com/deploy/).
+</aside>
+
+After this workflow-oriented run-through, let's see about different types of data you might want to model.
+
+## Neural networks for different kinds of data
+
+No introduction to deep learning without image classification. The "Fashion MNIST" classification tutorial mentioned in the beginning is a good introduction, but it uses a fully connected neural network to make it easy to remain focused on the overall approach. Standard models for image recognition, however, are commonly based on a convolutional architecture. [Here](https://tensorflow.rstudio.com/tutorials/advanced/images/cnn/) is a nice introductory tutorial.
+
+For text data, the concept of _embeddings_ -- distributed representations endowed with a measure of similarity -- is central. As in the aforementioned text classification tutorial, embeddings can be learned using the respective Keras layer; in fact, the more idiosyncratic the dataset, the more recommendable this approach. Often though, it makes a lot of sense to use _pre-trained embeddings_, obtained from large language models trained on enormous amounts of data. With TensorFlow Hub, discussed in more detail in the last section, pre-trained embeddings can be made use of simply by integrating an adequate _hub layer_, as shown in [one of the Hub tutorials](https://tensorflow.rstudio.com/tutorials/beginners/basic-ml/tutorial_basic_text_classification_with_tfhub/).
```

Comment:
```suggestion
For text data, the concept of _embeddings_ -- distributed representations endowed with a measure of similarity -- is central. As in the aforementioned text classification tutorial, embeddings can be learned using the respective Keras layers; in fact, the more idiosyncratic the dataset, the more recommendable this approach. Often though, it makes a lot of sense to use _pre-trained embeddings_, obtained from large language models trained on enormous amounts of data. With TensorFlow Hub, discussed in more detail in the last section, pre-trained embeddings can be made use of simply by integrating an adequate _hub layer_, as shown in [one of the Hub tutorials](https://tensorflow.rstudio.com/tutorials/beginners/basic-ml/tutorial_basic_text_classification_with_tfhub/).
```

Reply:
Changed "Keras layer" to "Keras layers" — but, obviously, change it back if I'm wrong about it referencing multiple layers. 🙂

Reply:
> With TensorFlow Hub, discussed in more detail in the last section,

This is one of those weird scenarios where "last" could refer to previous _or_ final. I think you mean final here, so you might switch it to that to ensure no one thinks they missed something earlier.

Reply:
yeah it was referring to one specific layer type - probably i should just name it, in parens!

Reply:
exactly... i knew this sounded weird, but "final" didn't come to mind...